ecs-agent/tmds/handlers/fault/v1/handlers/handlers_test.go

```diff
@@ -1017,37 +964,56 @@ func generateNetworkLatencyTestCases(name, expectedHappyResponseBody string) []n
 			},
 			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("required parameter Sources is missing"),
 			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
-				agentState.EXPECT().GetTaskMetadata(endpointId).
-					Return(state.TaskResponse{}, nil).
-					Times(0)
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{}, nil).Times(0)
+			},
+		},
+		{
+			name:               fmt.Sprintf("%s incomplete request body 3", name),
+			expectedStatusCode: 400,
+			requestBody: map[string]interface{}{
+				"Sources": ipSources,
+			},
+			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("required parameter DelayMilliseconds is missing"),
+			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{}, nil).Times(0)
 			},
 		},
 		{
 			name:               fmt.Sprintf("%s incomplete request body 3", name),
 			expectedStatusCode: 400,
 			requestBody: map[string]interface{}{
 				"DelayMilliseconds": delayMilliseconds,
-				"Sources":           ipSources,
+				"Sources":           []string{},
 			},
 			expectedResponseBody: types.NewNetworkFaultInjectionErrorResponse("required parameter JitterMilliseconds is missing"),
 			setAgentStateExpectations: func(agentState *mock_state.MockAgentState) {
-				agentState.EXPECT().GetTaskMetadata(endpointId).
-					Return(state.TaskResponse{}, nil).
-					Times(0)
+				agentState.EXPECT().GetTaskMetadata(endpointId).Return(state.TaskResponse{}, nil).Times(0)
 			},
 		},
 		{
-			name:               fmt.Sprintf("%s incomplete request body 4", name),
+			name:               fmt.Sprintf("%s invalid LossPercent in the request body 1", name),
```

Comment:
is this for latency fault?

Reply:
Thanks for the catch! The base structure of this method was copied over from the latency test cases, and I forgot to change the name here. Addressing in the next revision